### ISSUE TITLE
training vgg model E

### ISSUE THREAD

**Issue by pyadolla** [facebook/fbcunn#75]
I haven't had any luck training vgg model E from scratch using the example fbcunn code. After 20 epochs with a learning rate of 1e-2 (mom=.9, wd=5e-4), 10K epoch size, and 128 batch size on 4 Tesla K40's, I don't get any increase in training/test set accuracies or noticeable reduction in training loss. This seems very strange. I have not made any modifications to the example code and I have recent torch, nn, cunn, cudnn, and fbcunn modules. I don't experience any memory limitation errors using 128 batch size.

I've successfully trained alexnet with this code so I am wondering if anyone else has had luck with training vgg model E (or even model A) using the example code. The original paper states that they were successful with training model E not only by initializing some layers with model A pre-trained weights but /also/ by initializing model E weights using Xavier initialization which is what the nn code does.

**Comment by soumith** [comment]
If you want to train vgg model E from scratch using this code, you'll have to initialize the weights to MSR initialization.

Here's some code to do initialize weights according to different papers: https://github.com/e-lab/torch-toolbox/tree/master/Weight-init

The MSR initialization is referred to as "kaiming":
https://github.com/e-lab/torch-toolbox/blob/master/Weight-init/weight-init.lua#L43

**Comment by pyadolla** [comment]
Thanks a bunch, I'll give it a go.
